backend/src/main/java/com/github/EazyErik/controller/CustomDrinksController.java

```diff
@@ -0,0 +1,38 @@
+package com.github.EazyErik.controller;
+
+
+import com.github.EazyErik.datalayer.CustomDrink;
+import com.github.EazyErik.service.CustomDrinksService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+import java.util.List;
+
+@RestController
+@RequestMapping("api/customDrink")
+@RequiredArgsConstructor
+public class CustomDrinksController {
+
+    private final CustomDrinksService customDrinksService;
+
+
+
+    @PostMapping()
+    public CustomDrink addCustomDrink(@RequestBody CustomDrink customDrink, Principal username) {
+        return customDrinksService.addCustomDrink(customDrink,username.getName());
+
+    }
+    @GetMapping()
+    public List<CustomDrink> getCustomDrinks( @RequestParam(required = false) String ingredient) {
+        return customDrinksService.getCustomDrinks(ingredient);
+    }
+
+    @GetMapping("/details")
+    public CustomDrink getCustomDrink(@RequestParam String id) {
+        return customDrinksService.getCustomDrink(id);
+    }
```

Comment:
3/3
Here would the null Object end, return nothing or even crash. With an Optional you could use a ResponseEntity, which will do the check if a drink was found for you.